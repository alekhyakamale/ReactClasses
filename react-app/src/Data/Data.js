import { useField, Field } from "formik";
import React from "react";

export  const initialValues = [{
  email: "",
  loan: "",
  other_loan: "",
  addToLoan: "",
  balance: "",
  living: "",
  living_more: "",
  residence: "",
  dependents: "",
  dependents_more: "",
  dob: "",
  salary: "",
  incomeper: "",
  afterTax: false,
  allowances: "",
  rental: "",
  additional: "",
  utilities: "",
  utilitiesPer: "Month",
  household: "",
  householdPer: "Month",
  tv: "",
  tvPer: "Month",
  other: "",
  otherPer: "Month"
}]
export const MyInput = ({ label, ...props }) => {
  const [field, meta] = useField(props);
  return (
    <>
      <p className="label-styling" htmlFor={props.id || props.name}>{label}</p>
      <input className="mx-3" {...field} {...props} handlechange={props.handlechange}
            handleblur={props.handleblur} />
      {meta.touched && meta.error ? (
        <div className="error">{meta.error}</div>
      ) : null}
    </>
  );
};

export const ExpenseField = ({ name, label, ...props }) => {
  return (
    <>
      <div className="expense-row">
      <p className="label-styling">{label}</p>
        <div className="expense-row--left mx-4">
          <MyInput
            id={"expenses-" + name}
            type="number"
            label={""}
            name={name}
          />
        </div>
        {"  /  "}
        <div className="expense-row--right px-3">
          <Dropdown
            name={name + "Per"}
            selected_value={props.value || ["Month"]}
            handlechange={props.handlechange}
            handleblur={props.handleblur}
            options={["Week", "Fortnight", "Month", "Quarter", "Year"]}
          />
        </div>
      </div>
    </>
  );
};

export const CustomRadioGroup = ({ paragraph, label, ...props }) => {
  return (
    <Field
      key={props.id || props.name}
      name={props.name}
      component="div"
      render={({ field, meta }) => (
        <>
          <p className="label-styling" htmlFor={props.id || props.name}>{label}</p>
          <div className="radio-buttons">
            {props.options.map(opt => (
              <>
                <input
                  {...field}
                  type="radio"
                  id={props.name + opt}
                  name={props.name}
                  value={opt}
                  checked={opt === props.value}
                  handlechange={props.handlechange}
                  handleblur={props.handleblur}
                />
                <label className="hover-magic" htmlFor={props.name + opt}>{opt}</label>
              </>
            ))}
            {meta.touched && meta.error ? (
              <div className="error">{meta.error}</div>
            ) : null}
          </div>
        </>
      )}
    />
  );
};

export const Dropdown = ({ label, ...props }) => {
  return (
    <Field
      name={props.name}
      component="select"
      render={({ meta }) => (
        <>
          <p className="label-styling" htmlFor={props.id || props.name}>{label}</p>
          <div>
            <select
              name={props.name}
              value={props.selected_value}
              onChange={props.handlechange}
              onBlur={props.handleblur}
              style={{ display: "block" }}
            >
              {props.options.map(opt => (
                <option value={opt} label={opt} />
              ))}
            </select>

            {meta.touched && meta.error ? (
              <div className="error">{meta.error}</div>
            ) : null}
          </div>
        </>
      )}
    />
  );
};