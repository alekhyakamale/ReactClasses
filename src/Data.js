import { Field, useField } from "formik";
import React from "react";

export const MyInput = ({ span, label, ...props }) => {
    const [field, meta] = useField(props)
    return (
    <>
      <label htmlFor={props.id || props.name}>{label}</label>
      <span htmlFor={props.id || props.name}> {span}</span>
      <input type={props.type} {...field}/>
      {meta.error ? <div className="error">{meta.error}</div> : null}
    </>
  );
};

export const ExpenseField = ({ name, label, ...props }) => {
    return (
    <>
      <h3 style={{ marginBottom: 0 }}>{label}</h3>
      <div className="expense-row">
        <div className="expense-row--left">
          <MyInput
            id={"expenses-" + name}
            type="number"
            label={""}
            name={name}
          />
          {"  /  "}
        </div>
        <div className="expense-row--right">
          <Dropdown
            name={"expenses-" + name + "-period"}
            options={["Week", "Fortnight", "Month", "Quarter", "Year"]}
          />
        </div>
      </div>
    </>
  );
};

export const CustomRadioGroup = ({ paragraph, label, ...props }) => {
  return (
    <Field
      name={props.name}
      component="div"
      render={({ field, meta }) => (
        <>
          <h3 htmlFor={props.id || props.name}>{label}</h3>
          <p htmlFor={props.id || props.name}>{paragraph}</p>
          <div className="radio-buttons">
            {props.options.map(opt => (
              <>
                <input
                  {...field}
                  type="radio"
                  id={props.name + opt}
                  name={props.name}
                  value={opt}
                />
                <label for={props.name + opt}>{opt}</label>
              </>
            ))}
            {meta.error ? <div className="error">{meta.error}</div> : null}
          </div>
        </>
      )}
    />
  );
};

export const Dropdown = ({ label, ...props }) => {
  return (
    <Field
      name={props.name}
      component="div"
      render={() => (
        <>
          <h3 htmlFor={props.id || props.name}>{label}</h3>
          <div className="drop-down-menu">
            <select name={props.name}>
              {props.options.map(opt => (
                <>
                  <option name={props.name} value={opt}>
                    {opt}
                  </option>
                </>
              ))}
            </select>
          </div>
        </>
      )}
    />
  );
};