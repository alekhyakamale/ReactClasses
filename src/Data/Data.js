import { useField, Field } from "formik";
import React from "react";

export const MyInput = ({ label, ...props }) => {
  const [field, meta] = useField(props);
  return (
    <>
      <p className="label-styling" htmlFor={props.id || props.name}>{label}</p>
      <input className="mx-3" {...field} {...props} />
      {meta.touched && meta.error ? (
        <div className="error">{meta.error}</div>
      ) : null}
    </>
  );
};

export const ExpenseField = ({ name, label, ...props }) => {
  return (
    <>
      <div className="expense-row">
      <p className="label-styling">{label}</p>
        <div className="expense-row--left mx-4">
          <MyInput
            id={"expenses-" + name}
            type="number"
            label={""}
            name={name}
          />
        </div>
        {"  /  "}
        <div className="expense-row--right px-3">
          <Dropdown
            name={name + "Per"}
            selected_value={props.selected_period}
            handleChange={props.handleChange}
            handleBlur={props.handleBlur}
            options={["Week", "Fortnight", "Month", "Quarter", "Year"]}
          />
        </div>
      </div>
    </>
  );
};

export const CustomRadioGroup = ({ paragraph, label, ...props }) => {
  return (
    <Field
      name={props.name}
      component="div"
      render={({ field, meta }) => (
        <>
          <p className="label-styling" htmlFor={props.id || props.name}>{label}</p>
          <div className="radio-buttons">
            {props.options.map(opt => (
              <>
                <input
                  {...field}
                  type="radio"
                  id={props.name + opt}
                  name={props.name}
                  value={opt}
                  checked={opt === props.value}
                  handleChange={props.handleChange}
                  handleBlur={props.handleBlur}
                />
                <label className="hover-magic" for={props.name + opt}>{opt}</label>
              </>
            ))}
            {meta.touched && meta.error ? (
              <div className="error">{meta.error}</div>
            ) : null}
          </div>
        </>
      )}
    />
  );
};

export const Dropdown = ({ label, ...props }) => {
  return (
    <Field
      name={props.name}
      component="select"
      render={({ meta }) => (
        <>
          <p className="label-styling" htmlFor={props.id || props.name}>{label}</p>
          <div>
            <select
              name={props.name}
              value={props.selected_value}
              onChange={props.handleChange}
              onBlur={props.handleBlur}
              style={{ display: "block" }}
            >
              {props.options.map(opt => (
                <option value={opt} label={opt} />
              ))}
            </select>

            {meta.touched && meta.error ? (
              <div className="error">{meta.error}</div>
            ) : null}
          </div>
        </>
      )}
    />
  );
};