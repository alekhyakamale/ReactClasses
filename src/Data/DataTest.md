Test scenario: 
    Check whether all the input functionalities are working and the error 
    messages are getting displayed.

Test cases:
   1. Check the response on selecting an input in options of inputs.
   2. Check the response after entering a value in text fields.
   3. Check the response after selecting an option from the drop
        down menu.
   4. Check the response of error messages.

Test steps:
    1. Enter valid inputs in text fields and select a valid input in
        options fields.
    2. Enter a valid Date of birth.
    3. Check if error messages are getting displayed correctly by leaving
        the field blank or entering invalid inputs.
    4. Check whether the user inputs are getting cleared in between page 
        navigations.

Expected result:
    The form page should not show any errors and all fields must be filled.