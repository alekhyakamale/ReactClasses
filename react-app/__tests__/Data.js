import React from 'react';
import { CustomRadioGroup} from '../src/Data/Data';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import 'jest-extended';



configure({ adapter: new Adapter()})

test('renders Data without crashing', () => {
    const debug = shallow(<CustomRadioGroup/>);
    expect(debug.getElements()).toMatchSnapshot();
});