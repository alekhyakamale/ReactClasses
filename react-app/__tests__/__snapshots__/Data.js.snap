// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Data without crashing 1`] = `
Array [
  <Field
    component="div"
    render={[Function]}
  />,
]
`;
